jekyll/_cci2/creating-orbs.md

```diff
@@ -0,0 +1,332 @@
+* Parameter `steps` are powerful - Wrapping steps provided by the user allows you to encapsulate and sugar things like caching strategies and other more complex tasks, providing a lot of value to users.
+
+When developing your own orb, you may find it useful to write an inline orb. The section below describes how you can write your own inline orb.
+
+## Creating Inline Orbs
+
+Inline orbs can be handy during development of an orb or as a convenience for name-spacing jobs and commands in lengthy configurations, particularly if you later intend to share the orb with others.
+
+To write inline orbs, you need to place the orb elements under that orb's key in the ```orbs``` declaration in the configuration. For example, if you want to import one orb and then author inline for another, the orb might look like the example shown below:
+
+```
+orbs:
+  codecov: circleci/codecov-clojure@0.0.4
+  my-orb:
+    executors:
+      default:
+        docker:
+          - image: circleci/ruby:1.4.2
+    commands:
+      dospecialthings:
+        steps:
+          - run: echo "We will now do special things"
+    jobs:
+      myjob:
+        executor: default
+        steps:
+          - dospecialthings
+          - codecov/upload:
+              path: ~/tmp/results.xml
+
+version: 2.1
+workflows:
+  main:
+    jobs:
+      - my-orb/myjob
+```
+
+<aside class="notice">
+In the example above, note that the contents of ```my-orb``` are resolved as an inline orb because the contents of ```my-orb``` are a map; whereas the contents of ```codecov``` are a scalar value, and thus assumed to be an orb URI.
+</aside>
+
+### Example Template
+
+When you want to author an orb, you may wish to use this example template to quickly and easily create a new orb with all of the required concepts. This example includes each of the three top-level concepts of orbs. While any orb can be equally expressed as an inline orb definition, it will generally be simpler to iterate on an inline orb and use ```circleci config process .circleci/config.yml``` to check whether your orb usage matches your expectation.
+```
+version: 2.1
+orbs:
+  inline_example:
+    jobs:
+      my_inline_job:
+        parameters:
+          greeting_name:
+            type: string
+            default: olleh
+        executor: my_inline_executor
+        steps:
+          - my_inline_command:
+              name: <<parameters.greeting_name>>
+    commands:
+      my_inline_command:
+        parameters:
+          name:
+            type: string
+        steps:
+          - run: echo "hello <<parameters.name>>, from the inline command"
+    executors:
+      my_inline_executor:
+        parameters:
+          version:
+            type: string
+            default: "2.4"
+        docker:
+          - image: circleci/ruby:<<parameters.version>>
+
+workflows:
+  build-test-deploy:
+    jobs:
+      - inline_example/my_inline_job:
+          name: build
```

Comment:
It's worth calling out that you've assigned a name to these jobs. dnephin ran into issues earlier today because he tried to use `inline_example/my_inline_job` without assigning it a `name`. The config will compile to 2.0 config successful, but the workflow conductor doesn't know what to do with a job name that has a slash in it.

Reply:
what change do you want me to make to call this out?

Reply:
added a comment